.circleci/config.yml

```diff
@@ -22,7 +21,49 @@ jobs:
           paths:
             - ~/.m2
           key: v1-dependencies-{{ checksum "build.gradle" }}
+  run-api-diff:
+    steps:
+      - checkout
+      - run: chmod +x gradlew
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "build.gradle" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
```

Comment:
all these can be part of a single reusable step. Then build and test would run `clean check` & upload coverage, and apiDiff would run `clean apiDiff`. It should simplify a bit the script.

Reply:
Updated to extract common steps.